finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Parameter.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.BufferWriter
+import language.implicitConversions
+
+trait Parameter {
```

Comment:
@roanta I usually only use `sealed` when declaring ADTs/GADTs, but using it to prevent extension of `Parameter` from outside this unit seems like a great idea. 

Anonymous instantiation is okay as long as it's in the same unit.